### PR TITLE
Accessibility/button test

### PR DESCRIPTION
**Pull request by huzaifaaak**
## Summary
This PR aims to add test's for button.
Snapshot test for PR #31001 . This would make sure `accessibilityState` is properly set.

## Changelog

[General] [Added] - Test's for button

## Test Plan

`npm test` to run the test's. 
Since the disabled prop of button has precedence over `accessibilityState.disabled` the test's will make sure it remains this way.

### CODE REVIEW

**Comment by huzaifaaak** [comment]
@lunaleaps @kacieb Can you please review this

File: Libraries/Components/__tests__/Button-test.js
Comment (lunaleaps) [inline Libraries/Components/__tests__/Button-test.js:21]: nit: Should we mention that `accessibilityState` should also be disabled here?

**Review by lunaleaps — COMMENTED** [review]
Small nit in terms of setting expectations, otherwise looks good!

**Review by kacieb — CHANGES_REQUESTED** [review]
Hey! Thanks so much for writing these tests!

They can be simplified a bit - see my comment on another pull request about the same thing [here](https://github.com/facebook/react-native/pull/31135#issuecomment-805289384).

You can follow the same test structure as [this commit](https://github.com/facebook/react-native/commit/f69e096bb4df67474351786f674b1bb1e42d3363).

**Comment by lunaleaps** [comment]
@huzaifaaak I'll wait for your response from @kacieb's comment before landing.

**Comment by huzaifaaak** [comment]
@lunaleaps I have updated the test structure

**Pull request closed without merging**
